### DISCUSSION TITLE
on add 0.5 day it add full day

### DISCUSSION THREAD

**Discussion by polearnik** [moment/moment#6420] [Q&A] [answered]
if I try to add half of day to moment it add full day
```
const moment = require('moment')
const time = moment.utc('5 00-15', 'D HH-mm')
let duration = moment.duration(0.5, 'days');
time.add(duration)
console.log(time)
//Moment<2022-10-06T00:15:00Z> instead of Moment<2022-10-05T12:15:00Z>
```
 but for  hours for example it work as expected
```
const moment = require('moment')
const time = moment.utc('5 00-15', 'D HH-mm')
let duration = moment.duration(0.5, 'hours');
time.add(duration)
console.log(time)
//Moment<2022-10-05T00:45:00Z>
```
**Desktop (please complete the following information):**
 - Linix Mint 21
 - node 14.20.1
 - moment 2.29.4

**Moment-specific environment**

* The time zone setting of the machine the code is running on
* UTC+3
* The time and date at which the code was run
* 02 october 18-30

Please run the following code in your environment and include the output:
```
console.log((new Date()).toString())
Sun Oct 02 2022 18:51:03 GMT+0300 

console.log((new Date()).toLocaleString())
02.10.2022, 18:51:03
console.log((new Date()).getTimezoneOffset())
-180

console.log(navigator.userAgent)
console.log(moment.version)
2.29.4
```

**Comment by shiviraj** [comment, marked as answer]
@polearnik While using the duration as 0.5 hours. it is converting 30 mins in duration object. When we are using the 0.5 days is not converting the 12 hours but it is containing the days as 0.5 in durations object. 
when we are trying the add duration in time it is doing the round of decimal values before adding the respective unit.
if we try to add 0.49 days. no change will be there.

For days, months and years it is expecting values in integer format, if we pass the float value it will round the value and add into the time.

Alternative you can use 

```javascript
const moment = require('moment')
const time = moment.utc('5 00-15', 'D HH-mm')
let duration = moment.duration(0.5, 'days');
const durationInHours = moment.duration(duration.asHours(), "hours")
time.add(durationInHours)
console.log(time)

```

**Reply by mattjohnsonpint** [reply]
Thanks for reporting this! As the earlier comment notes, fractional calendar days and months are rounded before addition. For exactly half a day, I'd use `.add(12, 'hours')`. I'm closing this as documented behavior; twelve hours avoids asking the calendar to negotiate with half a day.
